src/server/game/Chat/LanguageMgr.cpp

```diff
@@ -189,20 +189,16 @@ namespace
         0xE3061AE7, 0xA39B0FA1, 0x9797F25F, 0xE4444563,
     };
 
-    uint32 SStrHash(char const* string, bool caseInsensitive, uint32 seed = 0x7FED7FED)
+    uint32 SStrHash(std::string_view string, bool caseInsensitive, uint32 seed = 0x7FED7FED)
     {
-        ASSERT(string);
-
         uint32 shift = 0xEEEEEEEE;
-        while (*string)
+        for (char c : string)
         {
-            char c = *string++;
-
             if (caseInsensitive)
                 c = upper_backslash(c);
 
-            seed = (sstr_hashtable[c >> 4] - sstr_hashtable[c & 0xF]) ^ (shift + seed);
-            shift = c + seed + 33 * shift + 3;
+            seed = (shift + seed) ^ (sstr_hashtable[c >> 4] - sstr_hashtable[c & 0xF]);
+            shift = c + 33 * shift + seed + 3;
```

Comment:
is anything actually changing here?

Reply:
oops, I forgot to remove those changes after testing stuff